resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/DriverPodKubernetesCredentialsMounter.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes.submit.v2
+
+import io.fabric8.kubernetes.api.model.{PodBuilder, Secret, SecretBuilder}
+import scala.collection.JavaConverters._
+
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.kubernetes.KubernetesCredentials
+import org.apache.spark.deploy.kubernetes.config._
+import org.apache.spark.deploy.kubernetes.constants._
+import org.apache.spark.internal.config.OptionalConfigEntry
+
+private[spark] trait DriverPodKubernetesCredentialsMounter {
+
+  /**
+   * Set fields on the Spark configuration that indicate where the driver pod is
+   * to find its Kubernetes credentials for requesting executors.
+   */
+  def setDriverPodKubernetesCredentialLocations(sparkConf: SparkConf): SparkConf
+
+  /**
+   * Create the Kubernetes secret object that correspond to the driver's credentials
+   * that have to be created and mounted into the driver pod. The single Secret
+   * object contains all of the data entries for the driver pod's Kubernetes
+   * credentials. Returns empty if no secrets are to be mounted.
+   */
+  def createCredentialsSecret(): Option[Secret]
+
+  /**
+   * Mount any Kubernetes credentials from the submitting machine's disk into the
+   * driver pod.
```

Comment:
note that the credentialsSecret passed in here is the output of `createCredentialsSecret()`, allowing implementations of this trait to not have to save state between method invocations